onnxruntime/core/providers/cpu/ml/svmclassifier.cc

```diff
@@ -125,120 +158,75 @@ Status SVMClassifier<T>::Compute(OpKernelContext* ctx) const {
 
           int64_t pos1 = (vector_count_) * (j - 1);
           int64_t pos2 = (vector_count_) * (i);
-          for (int64_t m = 0; m < class_i_support_count; m++) {
-            float val1 = coefficients_[pos1 + start_index_i + m];
-            float val2 = kernels[start_index_i + m];
-            sum += val1 * val2;
-          }
-          for (int64_t m = 0; m < class_j_support_count; m++) {
-            float val1 = coefficients_[pos2 + start_index_j + m];
-            float val2 = kernels[start_index_j + m];
-            sum += val1 * val2;
-          }
+          const float* val1 = &(coefficients_[pos1 + start_index_i]);
+          const float* val2 = &(kernels[start_index_i]);
+          for (int64_t m = 0; m < class_i_support_count; ++m, ++val1, ++val2)
+            sum += *val1 * *val2;
+
+          val1 = &(coefficients_[pos2 + start_index_j]);
+          val2 = &(kernels[start_index_j]);
+          for (int64_t m = 0; m < class_j_support_count; ++m, ++val1, ++val2)
+            sum += *val1 * *val2;
 
           sum += rho_[evals];
-          scores.push_back(sum);
-          if (sum > 0) {
-            votes[i]++;
-          } else {
-            votes[j]++;
-          }
-          evals++;  //index into rho
+          scores.push_back((float)sum);
+          ++(votes[sum > 0 ? i : j]);
+          ++evals;  //index into rho
         }
       }
-    } else if (mode_ == SVM_TYPE::SVM_LINEAR) {     //liblinear
-      for (int64_t j = 0; j < class_count_; j++) {  //for each class
-        float val = kernel_dot(x_data, current_weight_0, coefficients_, feature_count_ * j, feature_count_, get_kernel_type());
-        val += rho_[0];
-        scores.push_back(val);
-      }
     }
+
     if (proba_.size() > 0 && mode_ == SVM_TYPE::SVM_SVC) {
       //compute probabilities from the scores
-      std::vector<float> estimates;
-      std::vector<float> probsp2;
       int64_t num = class_count_ * class_count_;
-      for (int64_t m = 0; m < num; m++) {
-        probsp2.push_back(0.f);  //min prob
-      }
-      for (int64_t m = 0; m < class_count_; m++) {
-        estimates.push_back(0.f);  //min prob
-      }
+      std::vector<float> probsp2(num, 0.f);
+      std::vector<float> estimates(class_count_, 0.f);
       int64_t index = 0;
-      for (int64_t i = 0; i < class_count_; i++) {
-        for (int64_t j = i + 1; j < class_count_; j++) {
+      int64_t p1, p2;
```

Comment:
>int64_t p1, p2; [](start = 6, length = 15)

Uninitialized vars. This mostly likely will fail compilation.

Reply:
Fixed. Old habit from C.